temporalio/client.py

```diff
@@ -2508,10 +2514,14 @@ async def fetch_next_page(self, *, page_size: Optional[int] = None) -> None:
             page_size: Override the page size this iterator was originally
                 created with.
         """
+        page_size = page_size or self._input.page_size
+        if self._limit is not None and self._limit - self._yielded < page_size:
```

Comment:
The reason that we made the `fetch_next_page` public is because technically people don't have to use the async iteration, they can paginate using this themselves. For users using it that way, limit will not work. I am unsure if we want to make it work or if we want to say it's only when using it as an async iterable that it applies.

Reply:
Works for me. The reason behind making it public is I was afraid people wanted to use it outside of an `async for` and were afraid to invoke `__anext__` themselves and understand how to handle iteration stop and such.